### DISCUSSION TITLE
Optimize bundling excluding wordlists from viem/accounts

### DISCUSSION THREAD

**Discussion by tuler** [wevm/viem#2069] [Idea / Feature Request]
```typescript
import { mnemonicToAccount, privateKeyToAccount } from "viem/accounts";
```

The import above makes tree shaking add all `@scure/bip39/wordlists` packages, adding 226Kb to the bundle.
Wordlists are only needed for `generateMnemonic`.

**Comment by tmm** [comment]
What bundler are you using?

**Reply by tmm** [reply]
This should help, but still curious what setup you are using.

https://github.com/paulmillr/scure-bip39/pull/22

**Reply by tuler** [reply]
I'm using esbuild

**Reply by tuler** [reply]
Wrote a simple demo here https://github.com/tuler/viem-accounts-treeshaking

**Reply by tuler** [reply]
Any additional feedback on this?

**Reply by tmm** [reply]
Once https://github.com/paulmillr/scure-bip39/pull/22 is released, we can update the version and tree-shaking will be improved.

**Reply by tuler** [reply]
Any news after that was merged. I don't see any tree-shaking improvement with viem 2.21.1

**Reply by tmm** [reply]
It was just released in https://github.com/paulmillr/scure-bip39/releases/tag/1.4.0 yesterday so will update Viem shortly.

**Reply by tuler** [reply]
I upgraded my demo at https://github.com/tuler/viem-accounts-treeshaking to latest viem version 2.21.4 (and @scure/bip39@1.4.0), but I have the same results.

**Reply by jxom** [reply]
ESBuild doesn't support deep scope analysis, so unfortunately [it can't tree-shake these exports](https://github.com/wevm/viem/blob/f31e93ec5ad65e551bb3a152f138aff3a157b9f6/src/accounts/index.ts#L5-L14). [Pushed a commit](https://github.com/wevm/viem/commit/a5c56b8f4348cf6cc546b9f4c5280e69e411bb92) to fix.
